numba_cuda/numba/cuda/cudadrv/_info.py

```diff
@@ -0,0 +1,19 @@
+# Maps CTK version -> (min supported cc, max supported cc) inclusive
+_CUDA_CC_MIN_MAX_SUPPORT = {
```

Comment:
A drive-by comment, feel free to ignore. Hard-coding this list is not needed because it is queryable via NVRTC `nvrtcGetSupportedArchs` API: https://docs.nvidia.com/cuda/nvrtc/#group__query_1ga04c729958964c5e7b889a644f086ad4a, which is exposed in cuda-bindings.

Reply:
This API is added after 11.2: https://docs.nvidia.com/cuda/archive/11.2.2/cuda-toolkit-release-notes/index.html#cuda-general-new-features. 

I think it's a good idea to depend on this - feels more robust than hard coding as documentation can have bugs (like [this](https://github.com/NVIDIA/numba-cuda/pull/189#discussion_r2032076810)). This already covers a majority of the the CTK enlisted here, except from 11.1, which we manually added. cc @bdice

Reply:
Unrelated note - I think it would be great to add supported arch query API for libnvvm as well.

Reply:
I'm late to this PR, but it's OK to add this query at any point, because numba-cuda requires CUDA 11.2: https://nvidia.github.io/numba-cuda/user/overview.html#software

(it doesn't work with 11.1 and older because it has the older LLVM 3.4-based NVVM)